saleor/core/utils/__init__.py

```diff
@@ -130,3 +132,31 @@ def create_superuser(credentials):
     else:
         msg = 'Superuser already exists - %(email)s' % credentials
     return msg
+
+
+@python_2_unicode_compatible
+class TranslationWrapper(object):
+
+    def __init__(self, instance, locale):
+        self._instance = instance
+        self._locale = locale
+
+    @cached_property
+    def _translation(self):
+        return self._instance.translations.filter(language_code=self._locale).first()
```

Comment:
You can't call filter. Even if you put it behind ten thousand caches. It will execute an extra database query at least once per instance.